bundles/org.openhab.core.io.transport.modbus/src/main/java/org/openhab/core/io/transport/modbus/internal/pooling/ModbusSlaveConnectionFactoryImpl.java

```diff
@@ -292,6 +292,15 @@ private void tryConnect(ModbusSlaveEndpoint endpoint, PooledObject<ModbusSlaveCo
                             obj.getObject(), endpoint);
                 }
                 connection.connect();
+
+                logger.trace("Waiting {}ms for connection to warm up...", config.getAfterConnectionDelayMillis());
+                if (config != null && config.getAfterConnectionDelayMillis() > 0) {
```

Comment:
Seems like an unneccesary null check:

```suggestion
                if (config.getAfterConnectionDelayMillis() > 0) {
```